packit_service/worker/helpers/testing_farm_client.py

```diff
@@ -0,0 +1,278 @@
+
+
+class TestingFarmClient:
+    __test__ = False
+
+    def __init__(self, api_url: str, token: str, use_internal_tf: bool = False) -> None:
+        self.tft_api_url = api_url
+        if not self.tft_api_url.endswith("/"):
+            self.tft_api_url += "/"
+        self.tft_token = token
+        self.use_internal_tf = use_internal_tf
+        self.session = requests.session()
+        self.session.mount("https://", requests.adapters.HTTPAdapter(max_retries=5))
+
+    def get_raw_request(
+        self,
+        url: str,
+        method: str = "GET",
+        params: Optional[dict] = None,
+        data: Optional[dict] = None,
+    ) -> RequestResponse:
+        response = self.session.request(
+            method=method,
+            url=url,
+            params=params,
+            json=data,
+            timeout=HTTP_REQUEST_TIMEOUT,
+        )
+
+        try:
+            json_output = response.json()
+        except ValueError:
+            logger.debug(response.text)
+            json_output = None
+
+        return RequestResponse(
+            status_code=response.status_code,
+            ok=response.ok,
+            content=response.content,
+            json=json_output,
+            reason=response.reason,
+        )
+
+    def send_testing_farm_request(
+        self,
+        endpoint: str,
+        method: Optional[str] = None,
+        params: Optional[dict] = None,
+        data: Optional[dict] = None,
+    ) -> RequestResponse:
+        method = method or "GET"
+        url = f"{self.tft_api_url}{endpoint}"
+        try:
+            response = self.get_raw_request(
+                method=method,
+                url=url,
+                params=params,
+                data=data,
+            )
+        except requests.exceptions.ConnectionError as err:
+            logger.error(err)
+            raise PackitException(f"Cannot connect to url: `{url}`") from err
+        return response
+
+    def cancel_testing_farm_request(self, request_id: int) -> bool:
+        """
+        Cancel TF request with the given ID.
+
+        Args:
+            request_id: ID of the TF request.
+
+        Returns:
+            Whether the cancelling was successful.
+        """
+        logger.info(f"Cancelling TF request with ID {request_id}")
+        response = self.send_testing_farm_request(
+            endpoint=f"requests/{request_id}",
+            method="DELETE",
+        )
+        if response.status_code != 200:
+            msg = f"Failed to cancel TF request {request_id}: {response.json()}"
+            logger.error(msg)
+            return False
+        return True
+
+    @classmethod
+    def get_request_details(cls, request_id: str) -> dict[str, Any]:
+        """Testing Farm sends only request/pipeline id in a notification.
+        We need to get more details ourselves."""
+        service_config = ServiceConfig.get_service_config()
+        self = cls(
+            api_url=service_config.testing_farm_api_url,
```

Comment:
~~I guess it's coming from the original code, but could you add a comment here? It feels like we can't tell apart…~~ oh never mind, the API is public for both public and RH ranch :facepalm:

guess it deserves a comment anyways :D

Reply:
@mfocko Is this ok? https://github.com/packit/packit-service/blob/4dd28421c37345ad558fc9b5d653f4b2dcac7a0e/packit_service/worker/helpers/testing_farm_client.py#L113